src/puppet-dom.js

```diff
@@ -14,7 +14,10 @@
     var onDataReady = options.callback;
     this.element = options.listenTo || document.body;
     var clickHandler = this.clickHandler.bind(this);
-    var historyHandler = this.historyHandler.bind(this);
+    this.historyHandler = this.historyHandler.bind(this);
+
+    /* no need to wait for data to be ready to handle callback */
+    this.element.addEventListener('puppet-redirect-pushstate', this.historyHandler);
```

Comment:
@warpech I don't have any argument or reason against your solution tbh, but I just couldn't get it to work on my machine. I'll re-install starcounter and try it one more time. Because I played with `ClientFiles` a lot.